src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.QueueItem.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    internal partial class RequestExecutionQueue
+    {
+        private readonly struct QueueItem
+        {
+            public Func<RequestContext, Task<bool>> Callback { get; }
+            public bool MutatesSolutionState { get; }
+            public string? ClientName { get; }
+            public ClientCapabilities ClientCapabilities { get; }
```

Comment:
nit.  up to you.  but i find meaningless properties odd.  FOr basic internal structs, i woudl just make these readonly fields.